### PR TITLE
[Augmentation] Implement Breath of Eons module

### PR DESCRIPTION
**Pull request by Krealle**
### Description

Keeping in theme with rest of analysis implemented, I've implemented a new module to analysis the rotational performance of Breath of Eons usages. The module will analyze buff management around/during each individual Breath window, along with analyzing cast performance during the window. Providing feedback about Empower/trinket/potion usage, along with Ebon Might performance.

Visually there is a explanation section alongside with a graph for each window, explaning the performance of the individual windows.

Along with this is also a small fix to Ebon Might being applied/removed instead of refreshed.
Also added some missing trinkets.

### Testing

- Test report URL: 
 `/report/BYMXd4g3r6fJCtvQ/3-Mythic++The+Underrot+-+Kill+(28:14)/Ququlcoatl/standard/overview`
 `/report/TmF7aCGXy3ANzfDW/16-Mythic+Scalecommander+Sarkareth+-+Wipe+12+(4:07)/27-Dragoproject/standard`
- Screenshot(s):
image
image
image
image

### CODE REVIEW

File: src/analysis/retail/evoker/augmentation/modules/normalizers/EbonMightNormalizer.ts [hunk 1]
```diff
@@ -20,6 +30,9 @@ class EbonMightNormalizer extends EventsNormalizer {
     const fixedEvents: any[] = [];
     events.forEach((event: AnyEvent, idx: number) => {
       const linkedEvents = event._linkedEvents?.find((x) => x.relation === EBON_MIGHT_BUFF_LINKS);
+      const removeApplyLinks = event._linkedEvents?.find(
+        (x) => x.relation === EBON_MIGHT_APPLY_REMOVE_LINK,
+      );
```
Comment (trevorm4) [inline src/analysis/retail/evoker/augmentation/modules/normalizers/EbonMightNormalizer.ts]: Maybe use `GetRelatedEvents` instead

File: src/analysis/retail/evoker/augmentation/modules/normalizers/CastLinkNormalizer.ts [hunk 2]
```diff
@@ -81,6 +92,48 @@ const EVENT_LINKS: EventLink[] = [
...
+  {
+    linkRelation: EBON_MIGHT_APPLY_REMOVE_LINK,
+    reverseLinkRelation: EBON_MIGHT_APPLY_REMOVE_LINK,
+    linkingEventId: [SPELLS.EBON_MIGHT_BUFF_EXTERNAL.id],
+    linkingEventType: [EventType.ApplyBuff],
+    referencedEventId: SPELLS.EBON_MIGHT_BUFF_EXTERNAL.id,
+    referencedEventType: [EventType.RemoveBuff],
+    anyTarget: true,
+    forwardBufferMs: 0,
+    backwardBufferMs: 0,
```
Comment (trevorm4) [inline src/analysis/retail/evoker/augmentation/modules/normalizers/CastLinkNormalizer.ts]: You can omit these if you're using a buffer of 0

File: src/analysis/retail/evoker/augmentation/modules/normalizers/EbonMightNormalizer.ts [hunk 3]
```diff
@@ -10,7 +10,17 @@ import SPELLS from 'common/SPELLS/evoker';
  * The solution used here is to grab these extra events and turn them into RefreshBuffEvents.
  *
  * The issue described above can be seen here:
- * https://www.warcraftlogs.com/reports/hXcYL7yBrMzKA4Qd/#fight=8&type=auras&pins=0%24Separate%24%23244F4B%24casts%240%240.0.0.Any%24173872654.0.0.Evoker%24true%240.0.0.Any%24false%24395152%5E0%24Separate%24%23909049%24casts%240%240.0.0.Any%24173872654.0.0.Evoker%24true%240.0.0.Any%24false%24403631&target=1&ability=395152&view=events&start=5145354&end=5162301 */
+ * https://www.warcraftlogs.com/reports/hXcYL7yBrMzKA4Qd/#fight=8&type=auras&pins=0%24Separate%24%23244F4B%24casts%240%240.0.0.Any%24173872654.0.0.Evoker%24true%240.0.0.Any%24false%24395152%5E0%24Separate%24%23909049%24casts%240%240.0.0.Any%24173872654.0.0.Evoker%24true%240.0.0.Any%24false%24403631&target=1&ability=395152&view=events&start=5145354&end=5162301
+ *
+ * Issue two:
```
Comment (trevorm4) [inline src/analysis/retail/evoker/augmentation/modules/normalizers/EbonMightNormalizer.ts]:
```suggestion
 * Issue 2:
```

**Review by trevorm4 — CHANGES_REQUESTED** [review]
Overall LGTM just a few things and we can merge it

File: src/analysis/retail/evoker/augmentation/modules/normalizers/EbonMightNormalizer.ts (hunk 1 above)
Comment (Krealle) [reply src/analysis/retail/evoker/augmentation/modules/normalizers/EbonMightNormalizer.ts]: Changed to HasRelatedEvent, defo cleaner solution.

**Review by trevorm4 — APPROVED** [review]

**Pull request merged**